src/test/c_api_tests.cpp

```diff
@@ -274,6 +274,50 @@ TEST(CAPIStatusTest, GetCodeAndDetails) {
 
 class CAPIInference : public ::testing::Test {};
 
+TEST(CAPIServerMetadata, Basic) {
+    OVMS_ServerMetadata* metadata = nullptr;
+    OVMS_Server* cserver = nullptr;
+    ASSERT_CAPI_STATUS_NULL(OVMS_ServerNew(&cserver));
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_GetServerMetadata(nullptr, &metadata), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_GetServerMetadata(cserver, nullptr), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NULL(OVMS_GetServerMetadata(cserver, &metadata));
+    const char* json;
+    size_t size;
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_SerializeMetadataToJson(nullptr, &json, &size), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_SerializeMetadataToJson(metadata, nullptr, &size), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NOT_NULL_EXPECT_CODE(OVMS_SerializeMetadataToJson(metadata, &json, nullptr), StatusCode::NONEXISTENT_PTR);
+    ASSERT_CAPI_STATUS_NULL(OVMS_SerializeMetadataToJson(metadata, &json, &size));
+    ASSERT_EQ(std::strcmp(json, "{\"name\":\"REPLACE_PROJECT_NAME\",\"version\":\"REPLACE_PROJECT_VERSION\"}"), 0);
```

Comment:
Minor: you can compare std::string instead to get better error message in case strings differ